tests/test_virtual_chassis.py

```diff
@@ -24,3 +26,96 @@ def test_connectivity(self, vct):
                 _, out = dv.runcmd(['sh', "-c", "ping -c 5 -W 0 -q %s" % ip])
                 print(out)
                 assert '5 received' in out
+
+    def test_voq_switch(self, vct):
+        """ Test VOQ switch objects configuration """
+        dvss = vct.dvss
+        for name in dvss.keys():
+            if not name.startswith("supervisor"):
+                print("VOQ Switch test for {}".format(name))
+                dvs = dvss[name]
+                #Get the config info
+                config_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+                metatbl = swsscommon.Table(config_db, "DEVICE_METADATA")
+
+                status, cfg_switch_id = metatbl.hget("localhost", "switch_id")
+                assert status, "Got error in getting switch_id from CONFIG_DB DEVICE_METADATA"
+
+                status, cfg_max_cores = metatbl.hget("localhost", "max_cores")
+                assert status, "Got error in getting max_cores from CONFIG_DB DEVICE_METADATA"
+                
+                cfgsptbl = swsscommon.Table(config_db, "SYSTEM_PORT")
+                cfgspkeys = cfgsptbl.getKeys()
+                sp_count = len(cfgspkeys)
+
+                asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+                tbl = swsscommon.Table(asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_SWITCH")
+                keys = list(tbl.getKeys())
+                switch_oid_key = keys[0]
+
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_TYPE")
+                assert status, "Got error while getting switch type"
+                assert value == "SAI_SWITCH_TYPE_VOQ", "Switch type is not VOQ"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_SWITCH_ID")
+                assert status, "Got error while getting switch id"
+                assert value == cfg_switch_id, "VOQ switch id is invalid"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_MAX_SYSTEM_CORES")
+                assert status, "Got error while getting max system cores"
+                assert value == cfg_max_cores, "Max system cores is invalid"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST")
+                assert status, "Got error while getting system port config list"
+                #Convert the spcfg string to dictionary
+                spcfg = ast.literal_eval(value)
+                assert spcfg['count'] == sp_count, "Number of systems ports configured is invalid"
+    
```

Comment:
Any reason we don't check the config of individual system port?

Reply:
No specific reason. To save testing time. I thought just reading and verifying a field in the system port config list will be enough to verify successful config and retrieval of system port config list. Since the VS SAI emulation does not do any real functionality we just want to verify config and retrieval. Since system port config list is one single attribute, I thought checking any field in the attribute structure will be good enough.
If it will be good to check all ports in the list, we can do that. Please let me know

Reply:
It seems the system port info is defined in default_config.json, instead of being generated from minigraph, in vs. Then I guess we don't really have too much to validate here. What you have should fine.